src/components/NoIncrement/NumberIncrement.js

```diff
@@ -0,0 +1,95 @@
+"use client";
+import { useState, useEffect, useRef } from "react";
+import { motion, useSpring } from "framer-motion";
+import styled from "styled-components";
+
+const NumberIncrementWrapper = styled.div`
+  display: flex;
+  font-family: var(--font-geist-mono);
+  font-size: 48px;
+  font-weight: bold;
+  color: var(--foreground);
+`;
+
+const DigitColumn = styled.div`
+  position: relative;
+  width: 1em;
+  height: 1.2em;
+  overflow: hidden;
+`;
+
+const Digit = styled(motion.div)`
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  text-align: center;
+  will-change: transform;
+//   hints to browsers how an element is expected to change
+`;
+
```

Comment:
add documentation for this component. refer: https://schof.co/writing-jsdoc-for-react-components/